plugins/org.eclipse.n4js/src/org/eclipse/n4js/resource/N4JSResourceDescriptionStrategy.java

```diff
@@ -118,17 +118,57 @@ public boolean createEObjectDescriptions(final EObject eObject, IAcceptor<IEObje
 		return false;
 	}
 
+	/**
+	 * Creates the additional user data map for elements of type {@link TClass}.
+	 *
+	 * @param immutableUserData
+	 *            Map with basic entries which should be used to initialize a new user data map. This map is immutable
+	 *            and can thus not be modified.
+	 * @param tClass
+	 *            The {@link TClass} element to create user data for.
+	 * @returns An immutable user-data map
+	 */
+	protected Map<String, String> createClassUserData(final Map<String, String> immutableUserData,
+			final TClass tClass) {
```

Comment:
1. this was factored out from method `#internalCreateEObjectDescription()`, so I'd place this new method directly below `#internalCreateEObjectDescription()`.
2. why protected? I can't find anyone who overrides this atm. Would private make sense?